src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -859,6 +996,45 @@ void ScatterUpdate::scatterNDUpdate(uint8_t *indices, uint8_t *update, uint8_t *
     });
 }
 
+template <typename DataType, typename KernelType>
+void ScatterUpdate::scatterNDUpdate(const MemoryPtr& mem_data,
+                                    const MemoryPtr& mem_indices,
+                                    const MemoryPtr& mem_updates,
+                                    const KernelType& kernel) {
+    OPENVINO_ASSERT(reduction_type != ScatterUpdate::Reduction::NONE, "The reduction should not be NONE.");
+    uint8_t* indices = mem_indices->getDataAs<uint8_t>();
+    DataType* update = mem_updates->getDataAs<DataType>();
+    DataType* dstData = mem_data->getDataAs<DataType>();
+    const auto& srcDataDim = getParentEdgeAt(DATA_ID)->getMemory().getStaticDims();
+    const auto& indicesDim = getParentEdgeAt(INDICES_ID)->getMemory().getStaticDims();
+    const auto indicesRank = indicesDim.size();
+
+    std::vector<size_t> srcBlockND = getBlockND(srcDataDim);
+    const auto k = indicesDim[indicesRank - 1];
+    size_t idxTupleNum = 1;
+    for (size_t ri = 0; ri < indicesRank - 1; ri++) {
+        idxTupleNum *= indicesDim[ri];
+    }
+    const auto sizeToUpdate = srcBlockND[k];
+    for (size_t tupleIdx = 0; tupleIdx < idxTupleNum; tupleIdx++) {
```

Comment:
Just for my understanding: are there any plans for additional optimization? Like addining parallelization or manual vectorization?

Reply:
I was experimenting with parallelization, however without locks it could loose some updates, and with locks performance was bad depending on inputs shapes. Probably it would require some adjustments do algorithm to allow for further optimizations, but I run out of time